mne/io/_digitization.py

```diff
@@ -427,8 +427,18 @@ def _make_dig_points(nasion=None, lpa=None, rpa=None, hpi=None,
     if dig_ch_pos is not None:
         try:  # use the last 3 as int if possible (e.g., EEG001->1)
             idents = []
-            for key in dig_ch_pos:
+            for key, value in dig_ch_pos.items():
                 _validate_type(key, str, 'dig_ch_pos')
+                _validate_type(value, (np.ndarray, list, tuple), 'dig_ch_pos')
+                if isinstance(value, (list, tuple)):
+                    value = np.array(value)
+                    dig_ch_pos[key] = value
+                if value.dtype == int or value.dtype == np.float32:
+                    value = value.astype(np.float64)
+                    dig_ch_pos[key] = value
```

Comment:
```suggestion
                _validate_type(value, (np.ndarray, list, tuple), 'dig_ch_pos')
                value = np.array(value, dtype=float)
                dig_ch_pos[key] = value
```

I don't think it's necessary to avoid a copy here to keep the code simple